pkg/controller/clusterdeprovisionrequest/clusterdeprovisionrequest_controller.go

```diff
@@ -183,6 +195,32 @@ func (r *ReconcileClusterDeprovisionRequest) Reconcile(request reconcile.Request
 		metricUninstallJobDuration.Observe(float64(jobDuration.Seconds()))
 		return reconcile.Result{}, nil
 	}
+
+	// Check if the job should be regenerated:
+	newJobNeeded := false
+	if existingJob.Annotations == nil {
+		newJobNeeded = true
+	} else if _, ok := existingJob.Annotations[jobHashAnnotation]; !ok {
+		// this job predates tracking the job hash, so assume we need a new job
+		newJobNeeded = true
+	}
+
+	if existingJob.Annotations[jobHashAnnotation] != jobHash {
```

Comment:
should be `else if`'d with the statement above so it benefits from the nil check